MystenLabs/sui
# indexer: make event commit atomic and append only

### problem and theory
I found the event indexer's throughput will go down along with growth of event table, one reason is that today upon event commit, we do de-dup like below, which will be more and more expensive along with table growth
```
.on_conflict((transaction_digest, event_sequence))
.do_nothing()
```
the reason that we need this de-dup is that, we commit event and event logs in 2 steps (so not atomic), first commit events, second commit event log. However indexer itself can stop / crash in between, thus to avoid writing duplicate events to event table, we want to de-dup here.

### solution in this PR
This PR merges the events and event_log table such that we only need one DB commit for events thus it is atomic, in that sense we do not have to count on DB de-dup and can do append only.

This should help if not resolve the event throughput issue.

### Test
- run indexer with local validator and make sure events can be populated
- stop indexer, check DB and make sure that last event indeed has next cursor info
- restart indexer and make sure that indexer can proceed with new events
